examples/sprite_batch/Batching.cpp

```diff
@@ -0,0 +1,265 @@
+                        if (event.key.code == sf::Keyboard::F)
+                        {
+                            batchMode = modes[i == 0 ? modeCount - 1 : i - 1];
+                        }
+                        else
+                        {
+                            batchMode = modes[i == modeCount - 1 ? 0 : i + 1];
+                        }
+
+                        break;
+                    }
+                }
+            }
+
+            // Window size changed, adjust view appropriately
+            if (event.type == sf::Event::Resized)
+            {
+                sf::View view;
+                view.setSize({gameWidth, gameHeight});
+                view.setCenter({gameWidth / 2.f, gameHeight / 2.f});
+                window.setView(view);
+            }
+        }
+
+        // Clear the window
+        window.clear(sf::Color(50, 50, 50));
+
+        // Delete excess sprites
+        while (spriteCount < sprites.size())
+        {
+            sprites.pop_back();
+        }
+
+        // Add new sprites
+        while (spriteCount > sprites.size())
+        {
+            sf::Vector2f       position = sf::Vector2f(static_cast<float>(rand() % gameWidth),
+                                                 static_cast<float>(rand() % gameHeight));
+            sf::Angle          rotation = sf::degrees(static_cast<float>(rand() % 360));
+            const sf::Texture* texture  = &textures[static_cast<std::size_t>(rand()) % textures.size()];
+
+            // X% chance to use the previous texture instead
+            // The ordered batcher performs badly when textures change extremely often.
+            if (sprites.size() > 0 && rand() % 100 < sameTextureChance)
+                texture = sprites[sprites.size() - 1].getTexture();
+
+            if (position.x > gameWidth)
+                position.x -= gameWidth;
+            if (position.x < 0)
+                position.x += gameWidth;
+
+            if (position.y > gameHeight)
+                position.y -= gameHeight;
+            if (position.y < 0)
+                position.y += gameHeight;
+
+            auto& sprite = sprites.emplace_back();
+
+            sprite.setPosition(position);
+            sprite.setRotation(rotation);
+            sprite.setTexture(*texture);
+        }
+
+        // Update sprites
+        for (auto& sprite : sprites)
+        {
+            sprite.setRotation(sprite.getRotation() + sf::degrees(2));
+        }
+
+        // Draw sprites
+        clock.restart();
+
+        float       depth = 1.00f;
+        std::size_t count = 0;
+
+        batch.begin(batchMode, window, nullptr, sf::BlendAlpha);
+
+        for (auto& sprite : sprites)
+        {
+            batch.draw(sprite, depth);
+            count++;
+
+            if (count % 250 == 0)
+                depth -= 0.01f;
+        }
+
+        batch.end();
+
+        auto time = clock.restart();
+
+        timeHistory.push_back(static_cast<float>(time.asMicroseconds()) / 1000.f);
```

Comment:
I've switched to `asSeconds()`, but I'm still doing `timeReference.asSeconds() * 1000.0f` since I want floating point milliseconds. Not sure if there's a better way to do it.

Reply:
Ideally you'd store `sf::Time` objects instead of storing floats. The whole point of `sf::Time` is that you don't have to weak, primitive types to store durations.

Reply:
Yeah, that is what I'm doing right now. However, I'm doing the conversion to display the values to window:
```
debugInfoText.setString(
            std::string(batchEnabled ? "true" : "false") + "\n" +
            getDisplayString(benchmarkedDrawable) + "\n" +
            std::to_string(timeReference.asSeconds() * 1000.0f) + "\n" +
            std::to_string(average.asSeconds() * 1000.0f) + "\n" +
            std::to_string(drawableCount) + "\n" +
            getDisplayString(batchMode) + "\n" +
            (useStaticBatch ? "Static batch is active" : ""));
```